tests/testapp/tests/test_syncsession_methods.py

```diff
@@ -0,0 +1,251 @@
+        for i in self.data['group1_c2']:
+            self.assertIn(i.id, buffer_ids)
+            self.assertIn(i.id, rmcb_ids)
+        for i in self.data['group2_c1']:
+            self.assertIn(i.id, buffer_ids)
+            self.assertIn(i.id, rmcb_ids)
+
+    def test_fsic_specific_id(self):
+        self.fsics = {self.data['group2_id'].id: 0}
+        self.data['sc']._queue_into_buffer(self.filter_prefixes, self.fsics)
+        buffer_ids = Buffer.objects.values_list('model_uuid', flat=True)
+        rmcb_ids = RecordMaxCounterBuffer.objects.values_list('model_uuid', flat=True)
+        for i in self.data['group1_c1']:
+            self.assertNotIn(i.id, buffer_ids)
+            self.assertNotIn(i.id, rmcb_ids)
+        for i in self.data['group1_c2']:
+            self.assertNotIn(i.id, buffer_ids)
+            self.assertNotIn(i.id, rmcb_ids)
+        for i in self.data['group2_c1']:
+            self.assertIn(i.id, buffer_ids)
+            self.assertIn(i.id, rmcb_ids)
+
+    def test_fsic_counters(self):
+        counter = InstanceIDModel.objects.get(id=self.data['group1_id'].id).counter
+        self.fsics = {self.data['group1_id'].id: counter - 1}
+        self.data['sc']._queue_into_buffer(self.filter_prefixes, self.fsics)
+        buffer_ids = Buffer.objects.values_list('model_uuid', flat=True)
+        rmcb_ids = RecordMaxCounterBuffer.objects.values_list('model_uuid', flat=True)
+        for i in self.data['group1_c1']:
+            self.assertNotIn(i.id, buffer_ids)
+            self.assertNotIn(i.id, rmcb_ids)
+        for i in self.data['group1_c2']:
+            self.assertIn(i.id, buffer_ids)
+            self.assertIn(i.id, rmcb_ids)
+        for i in self.data['group2_c1']:
+            self.assertNotIn(i.id, buffer_ids)
+            self.assertNotIn(i.id, rmcb_ids)
+
+    def test_fsic_counters_too_high(self):
+        self.fsics = {self.data['group1_id'].id: 100, self.data['group2_id'].id: 100}
+        self.data['sc']._queue_into_buffer(self.filter_prefixes, self.fsics)
+        self.assertFalse(Buffer.objects.all())
+        self.assertFalse(RecordMaxCounterBuffer.objects.all())
+
+    def test_partition_filter_buffering(self):
+        self.filter_prefixes = ['{}:user:summary'.format(self.data['user3'].id),
+                                '{}:user:interaction'.format(self.data['user3'].id)]
+        self.data['sc']._queue_into_buffer(self.filter_prefixes, self.fsics)
+        buffer_ids = Buffer.objects.values_list('model_uuid', flat=True)
+        rmcb_ids = RecordMaxCounterBuffer.objects.values_list('model_uuid', flat=True)
+        self.assertNotIn(self.data['user2'].id, buffer_ids)
+        self.assertNotIn(self.data['user2'].id, rmcb_ids)
+        for i in self.data['user3_sumlogs']:
+            self.assertIn(i.id, buffer_ids)
+            self.assertIn(i.id, rmcb_ids)
+        for i in self.data['user3_interlogs']:
+            self.assertIn(i.id, buffer_ids)
+            self.assertIn(i.id, rmcb_ids)
```

Comment:
Perhaps the reusable function I described above could be more general, along the lines of:

`assertRecordsBuffered(filter, fsics, records)`

which here would be:
```
filter_prefixes = ['{}:user:summary'.format(self.data['user3'].id), '{}:user:interaction'.format(self.data['user3'].id)]
self.assertRecordsBuffered(filter_prefixes, self.fsics, self.data['user3_sumlogs'] + self.data['user3_interlogs'])
```

That could cut down the verbosity and increase the expressiveness of these tests.